tests/opwen_email_server/utils/test_email_parser.py

```diff
@@ -289,3 +289,20 @@ def test_format_attachments_with_image(self):
 
         self.assertNotEqual(input_content, output_content)
         self.assertEqual(input_filename, output_filename)
+
+
+class DescendingTimestampTests(TestCase):
+    def test_descending_timestamp_correct(self):
+        sent_at = '2020-02-01 21:09'
+
+        sent_at_timestamp = email_parser.descending_timestamp(sent_at)
+
+        self.assertEqual(sent_at_timestamp, '519408660')
+
+    def test_descending_timestamp_ordering(self):
+        multiple_sent_at = ['2020-02-01 21:09', '2020-02-01 22:09']
+
+        sorted_email_timestamp = sorted(
+            [email_parser.descending_timestamp(date_time) for date_time in multiple_sent_at])
+
+        self.assertEqual(sorted_email_timestamp, ['519405060', '519408660'])
```

Comment:
Given that the code/idea behind the sorting isn't super obvious, I'd really encourage to make tests work as sort of "executable documentation" to communicate the intent behind the code, namely that we want to create a string ordering of dates that puts later dates before earlier dates.

As such, I would make this test even more explicit to make it super clear what our desired behavior is, perhaps something like this:

```suggestion
        january = '2020-01-01 22:09'
        february = '2020-02-02 22:09'

        january_timestamp = email_parser.descending_timestamp(january)
        february_timestamp = email_parser.descending_timestamp(february)
        
        timestamp_ordering = sorted([january_timestamp, february_timestamp])

        self.assertEqual(timestamp_ordering, [february_timestamp, january_timestamp])
```

We could then also add a few more tests of the same style (copy/paste/replace is fine) to test other time differences (e.g. years, days, hours, and minutes).

Reply:
Updated tests to reflect year, month, day, hour, minute. I included 'h' and 'm' into the time variables to reflect hour and minute. I felt that without it, there could be confusion between the years in previous test variable names and times.